src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -168,6 +169,9 @@ def collectAgentInfo(self):
             if float(disk['percent'].strip('%')) >= diskUseThreshold and \
                             disk['mounted'] not in self.config.AnalyticsDataCollector.ignoreDisk:
                 agentInfo['disk_warning'].append(disk)
+                agentInfo['drain_mode'] = True
+                self.config.WorkQueueManager.queueParams['DrainMode'] = True
+                os.system("$manage execute-agent wmcoreD --restart --components=WorkQueueManager,AnalyticsDataCollector,AgentStatusWatcher")
```

Comment:
@ticoann @amaltaro 
I'm not sure this is the best place to do the restart of these services?

Also I wanted to use the functions directly from the wmcoreD script however it is not in a module dir so not importable and is written with global variables which are populated from command line args and as such it's not easy to use the code directly. I don't know if you would have preferred spawning a subprocess with the subprocess module, or using a separate thread or an exec (if we want to replace the service anyway) for this or if a simple `os.system` is sufficient? I guess it really depends on the behaviour your looking for. I'll let you decide.